log4j-api/src/main/java/org/apache/logging/log4j/message/ReusableObjectMessage.java

```diff
@@ -100,16 +100,17 @@ public Throwable getThrowable() {
      */
     @Override
     public Object[] swapParameters(final Object[] emptyReplacement) {
+        emptyReplacement[0] = obj;
```

Comment:
I can't imagine that `emptyReplacement` should ever be empty, but it might be worth  doing something along these lines:

```java
if (emptyReplacement.length == 0) {
    Object[] params = new Object[10]; // Default reusable parameter buffer size
    params[0] = obj;
    return params;
}
emptyReplacement[0] = obj;
return emptyReplacement;
```

Reply:
@cakofony fixed